tests/planner/ompl/test_OMPLSimplifier.cpp

```diff
@@ -0,0 +1,189 @@
+#include "OMPLTestHelpers.hpp"
+#include "../../constraint/MockConstraints.hpp"
+#include <aikido/planner/ompl/Planner.hpp>
+#include <aikido/planner/ompl/CRRT.hpp>
+#include <aikido/planner/ompl/CRRTConnect.hpp>
+#include <aikido/trajectory/Interpolated.hpp>
+#include <aikido/constraint/uniform/RnBoxConstraint.hpp>
+#include <aikido/constraint/CartesianProductSampleable.hpp>
+#include <aikido/constraint/CartesianProductTestable.hpp>
+#include <aikido/constraint/JointStateSpaceHelpers.hpp>
+#include <aikido/util/StepSequence.hpp>
+#include <aikido/planner/ompl/MotionValidator.hpp>
+#include <ompl/geometric/planners/rrt/RRTConnect.h>
+
+using StateSpace = aikido::statespace::dart::MetaSkeletonStateSpace;
+using aikido::planner::ompl::getSpaceInformation;
+using aikido::planner::ompl::CRRT;
+using aikido::planner::ompl::CRRTConnect;
+
+namespace {
+
+  template <typename T>
+  double computeTrajLength(T& traj,
+      aikido::statespace::dart::MetaSkeletonStateSpacePtr stateSpace,
+      aikido::distance::DistanceMetricPtr& dmetric)
+  {
+    auto stateCurrent = stateSpace->createState();
+    auto stateNext = stateSpace->createState();
+
+    double trajDistance = 0.0;
+    
+    for(size_t i = 0; i < traj->getDuration(); ++i)
+    {
+      traj->evaluate(i, stateCurrent);   
+      traj->evaluate(i+1, stateNext);
+      trajDistance += dmetric->distance(stateCurrent, stateNext);
+    }
+
+    return trajDistance;
+  }
+
+  aikido::trajectory::InterpolatedPtr planTrajForTest(
+      aikido::statespace::dart::MetaSkeletonStateSpacePtr _stateSpace,
+      aikido::statespace::InterpolatorPtr& _interpolator,
+      aikido::distance::DistanceMetricPtr& _dmetric,
+      aikido::constraint::SampleablePtr& _sampler,
+      aikido::constraint::TestablePtr& _collConstraint,
+      aikido::constraint::TestablePtr& _boundsConstraint,
+      aikido::constraint::ProjectablePtr& _boundsProjection,
+      Eigen::Vector3d _startPose, Eigen::Vector3d _goalPose          
```

Comment:
It [is dangerous](https://eigen.tuxfamily.org/dox/group__TopicPassingByValue.html) to pass `Eigen` types by value because it can introduce alignment issues. Please pass these by `const &`.